src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_LENGTH = 10;
+    private int arraySize;
+    private T[] objects;
+
+    public ArrayList() {
+        objects = (T[]) (Array.newInstance(Object.class, START_ARRAY_LENGTH));
+    }
+
     @Override
     public void add(T value) {
-
+        if (objects.length == arraySize) {
+            objects = grow(objects);
+        }
```

Comment:
```suggestion
        grow(objects);
```

move validation to grow function since you have same validation several times